jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1115 @@
+            if wcs is not None:
+                raise GalSimIncompatibleValuesError(
+                    "Cannot provide both scale and wcs to InterpolatedImage",
+                    scale=self._jax_children[1]["scale"],
+                    wcs=self._jax_children[1]["wcs"],
+                )
+        elif wcs is not None:
+            if not isinstance(wcs, BaseWCS):
+                raise TypeError("wcs parameter is not a galsim.BaseWCS instance")
+        else:
+            if self._jax_children[0].wcs is None:
+                raise GalSimIncompatibleValuesError(
+                    "No information given with Image or keywords about pixel scale!",
+                    scale=self._jax_children[1]["scale"],
+                    wcs=self._jax_children[1]["wcs"],
+                    image=self._jax_children[0],
+                )
+
+    @doc_inherit
+    def withGSParams(self, gsparams=None, **kwargs):
+        if gsparams == self.gsparams:
+            return self
+        # Checking gsparams
+        gsparams = GSParams.check(gsparams, self.gsparams, **kwargs)
+        # Flattening the representation to instantiate a clean new object
+        children, aux_data = self.tree_flatten()
+        aux_data["gsparams"] = gsparams
+        ret = self.tree_unflatten(aux_data, children)
+
+        return ret
+
+    def tree_flatten(self):
+        """This function flattens the InterpolatedImage into a list of children
+        nodes that will be traced by JAX and auxiliary static data."""
+        return (self._jax_children, copy.copy(self._jax_aux_data))
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, children):
+        """Recreates an instance of the class from flatten representation"""
+        val = {}
+        val.update(aux_data)
+        val.update(children[1])
+        ret = cls(children[0], **val)
+        return ret
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        d.pop("_workspace")
+        return d
+
+    def __setstate__(self, d):
+        self.__dict__ = d
+        self._workspace = {}
+
+    @property
+    def x_interpolant(self):
+        """The real-space `Interpolant` for this profile."""
+        return self._x_interpolant
+
+    @property
+    def k_interpolant(self):
+        """The Fourier-space `Interpolant` for this profile."""
+        return self._k_interpolant
+
+    @lazy_property
+    def image(self):
+        """The underlying `Image` being interpolated."""
+        return self._xim[self._image.bounds]
+
+    @property
+    def _flux(self):
+        return self._image_flux
+
+    @lazy_property
+    def _centroid(self):
+        x, y = self._pad_image.get_pixel_centers()
+        tot = jnp.sum(self._pad_image.array)
+        xpos = jnp.sum(x * self._pad_image.array) / tot
+        ypos = jnp.sum(y * self._pad_image.array) / tot
+        return PositionD(xpos, ypos)
+
+    @lazy_property
+    def _max_sb(self):
+        return jnp.max(jnp.abs(self._pad_image.array))
```

Comment:
does it matter that you do this in the `_pad_image` rather than on the `_image` directly?

Reply:
You want to use pad image so you get the maximum SB of any non-zero pixels IIUIC.